tests/testthat/test-run-nmtran.R

```diff
@@ -0,0 +1,261 @@
+
+        # Override with model yaml
+        current_args <- mod1$bbi_args
+        mod1 <- add_bbi_args(mod1, list(tprdefault = TRUE, maxlim = 3))
+
+        # Test priority of .bbi_args > model yaml > bbi.yaml
+        # - TRUE values overwrite FALSE values if both are present
+
+        # Overwrite .bbi_args --> takes precedence
+        cmd_args <- mod1 %>% parse_nmtran_args(
+          .bbi_args = list(prdefault = TRUE, tprdefault = FALSE, maxlim = 1),
+          nmfe_options = bbi_yaml$nmfe_options
+        )
+
+        expect_equal(unname(cmd_args), c("1", "0", "1"))
+
+        # Overwrite bbi.yaml --> model yaml defaults take precedence
+        cmd_args <- mod1 %>% parse_nmtran_args(
+          nmfe_options = list(prdefault = TRUE, tprdefault = FALSE, maxlim = 1)
+        )
+
+        expect_equal(unname(cmd_args), c("1", "1", "3"))
+
+        # Check with all three sources
+        cmd_args <- mod1 %>% parse_nmtran_args(
+          .bbi_args = list(tprdefault = FALSE, maxlim = 2),
+          nmfe_options = list(tprdefault = FALSE, maxlim = 1)
+        )
+
+        expect_equal(unname(cmd_args), c("0", "0", "2"))
+
+        cmd_args <- mod1 %>% parse_nmtran_args(
+          .bbi_args = list(prdefault = TRUE, tprdefault = FALSE, maxlim = 2),
+          nmfe_options = list(tprdefault = TRUE, maxlim = 1)
+        )
+
+        expect_equal(unname(cmd_args), c("1", "1", "2"))
+
+        # Reset model yaml .bbi_args for other tests
+        mod1 <- replace_all_bbi_args(mod1, current_args)
+      })
+
+      it("execute_nmtran", {
+        # Execute in subdirectory to avoid messing with other tests
+        nmtran_dir <- file.path(MODEL_DIR_BBI, "nmtran")
+        fs::dir_create(nmtran_dir)
+        on.exit(fs::dir_delete(nmtran_dir), add = TRUE)
+
+        # Copy model file into new model dir
+        fs::file_copy(CTL_TEST_FILE, file.path(nmtran_dir, "2.ctl"))
+        mod2 <- new_model(file.path(nmtran_dir, "2"))
+
+        # create new bbi.yaml
+        bbi_init(
+          nmtran_dir,
+          .nonmem_dir = NONMEM_DIR,
+          .nonmem_version = "nm74gf",
+          .bbi_args = list(mpi_exec_path = get_mpiexec_path())
+        )
+
+        nmtran_specs <- nmtran_setup(mod2)
+
+        nmtran_results <- execute_nmtran(
+          nmtran_specs$nmtran_exe, mod_path = get_model_path(mod2),
+          dir = nmtran_dir
+        )
+
+        # Check attributes
+        expect_equal(nmtran_dir, nmtran_results$run_dir)
+        expect_equal(nmtran_results$status_val, 0)
+        expect_equal(nmtran_results$status, "Finished Running")
+
+        # Test failure
+        data_path <- "test/this/path/data.csv"
+        modify_data_path_ctl(mod2, data_path)
+
+        nmtran_results <- execute_nmtran(
+          nmtran_specs$nmtran_exe, mod_path = basename(get_model_path(mod2)),
+          dir = nmtran_dir
+        )
+
+        # Check attributes
+        expect_equal(nmtran_results$status_val, 4)
```

Comment:
Okay as is, but I'd suggest just checking that it's greater than zero rather than that it's a particular non-zero value. Using a particular value is unnecessarily coupled to the details of NM-TRAN (unless we have a reason to expect a specific non-zero value).